### PR TITLE
tiledb 2.26.2

### PR DESCRIPTION
**Pull request by ifitchet**
tiledb 2.26.2

**Destination channel:** Defaults

### Links

- [PKG-5641]
- dev_url:        https://github.com/TileDB-Inc/TileDB/tree/2.26.2
- conda_forge:    https://github.com/conda-forge/tiledb-feedstock
- pypi:           n/a
- pypi inspector: n/a

### Explanation of changes:

- new version number
- it was decided to skip windows entirely for this build as this only goes to snowflake channel

### tiledb-patches directory
Explained in detail here: https://github.com/conda-forge/tiledb-feedstock/tree/main/recipe/tiledb-patches/system-ports
TLDR: This tells vcpkg to not vendor (download and build) certain dependencies that we have already via conda.

[PKG-5641]: https://anaconda.atlassian.net/browse/PKG-5641?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

EDIT: changed dev_url to match version

### CODE REVIEW

File: recipe/0002-Don-t-link-to-fmt-library.-We-use-spdlog-s-embedded-.patch [hunk 1]
```diff
@@ -1,32 +1,44 @@
-From f9733cf7f72f3997566679f37641a3240392e05f Mon Sep 17 00:00:00 2001
-From: Jean-Christophe Morin <jcmorin@anaconda.com>
```
Comment (JeanChristopheMorinPerso) [inline recipe/0002-Don-t-link-to-fmt-library.-We-use-spdlog-s-embedded-.patch:2]: You should keep the header. You can put your name if you want, but we should keep the header to know where it's coming from.

File: recipe/bld.bat [hunk 2]
```diff
@@ -1,26 +1,32 @@
...
+      -DVCPKG_CMAKE_CONFIGURE_OPTIONS=-DCMAKE_FIND_DEBUG_MODE=TRUE ^
       ..
 if errorlevel 1 exit 1
 
-nmake
-if errorlevel 1 exit 1
-
-pushd "tiledb"
-nmake install
+cmake --build . -j --target install
```
Comment (JeanChristopheMorinPerso) [inline recipe/bld.bat:30]: I don't think `-j` is necessary...

File: recipe/meta.yaml [hunk 3]
```diff
@@ -19,32 +17,44 @@ build:
   skip: true  # [win and vc<14]
   missing_dso_whitelist:  # [linux and s390x]
     - $RPATH/ld64.so.1    # [linux and s390x]
+  script_env:
+    # "Environment variable VCPKG_FORCE_SYSTEM_BINARIES must be set on
+    # arm, s390x, ppc64le and riscv platforms."
+    - VCPKG_FORCE_SYSTEM_BINARIES=1  # [ arm64 or aarch64 or s390x or ppc64le]
+  overlinking_ignore_patterns:
+    # complains about libfmt.so
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:25]: We should investigate. If it complains, it's usually for good reasons.

File: recipe/meta.yaml [hunk 4]
```diff
@@ -19,32 +17,44 @@ build:
...
-    - m2-patch  # [win]
   host:
     - zlib {{ zlib }}
     - bzip2 {{ bzip2 }}
     - zstd {{ zstd }}
-    - lz4-c 1.9.4
-    - curl 7.88.1 # [not win]
-    - libcurl 7.88.1  # [not win]
-    - openssl  {{ openssl }}  # [not win]
+    - lz4-c
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:43]: We should probably keep these pinned.

File: recipe/meta.yaml [hunk 5]
```diff
@@ -76,3 +86,5 @@ extra:
     - shelnutt2
     - ihnorton
     - nguyenv
+  skip-lints:
+    - no_git_on_windows
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:90]: This should not be necessary

File: recipe/meta.yaml [hunk 6]
```diff
@@ -19,32 +17,44 @@ build:
...
+
 requirements:
   build:
-    - make  # [not win]
-    - git   # [not win]
-    - sed   # [not win]
-    - gawk  # [not win]
-    - cmake <=3.21
-    - pkg-config  # [not win]
+    - make                   # [not win]
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:30]: We could use ninja on all platforms.

File: recipe/0002-Don-t-link-to-fmt-library.-We-use-spdlog-s-embedded-.patch (hunk 1 above)
Comment (Cansisti) [reply recipe/0002-Don-t-link-to-fmt-library.-We-use-spdlog-s-embedded-.patch:2]: I think this patch was regenerated by Ian using `git diff`, thus no header.
We are not using it anymore tho, and I don't know what commit he used for it.

File: recipe/bld.bat (hunk 2 above)
Comment (Cansisti) [reply recipe/bld.bat:30]: It's quite large project, going single thread will take significant time to compile.
Is there any reason to not use `-j`?

File: recipe/meta.yaml (hunk 3 above)
Comment (Cansisti) [reply recipe/meta.yaml:25]: I guess that's what the patch was for, but it failed. I will give it a try.

File: recipe/meta.yaml (hunk 5 above)
Comment (Cansisti) [reply recipe/meta.yaml:90]: git is used by vcpkg to download it's source.
I've decided to leave vcpkg enabled on windows (ie not unvendor it) as disabling it (and removing git) would require building libmagic (or file-feedstock) for windows.
They are doing it internally with some patches to libmagic source code that make is possible:
https://github.com/TileDB-Inc/TileDB/tree/2.26.2/ports/libmagic
Generally, I agree it's not the cleanest solution, but delivering it became more of a priority than cleanness.

File: recipe/meta.yaml (hunk 6 above)
Comment (Cansisti) [reply recipe/meta.yaml:30]: I'll try it.

File: recipe/bld.bat (hunk 2 above)
Comment (skupr-anaconda) [reply recipe/bld.bat:30]:
```suggestion
cmake --build . -j %CPU_COUNT% --target install
```
Missing %CPU_COUNT%

File: recipe/build.sh [hunk 7]
```diff
@@ -27,21 +31,58 @@ if [[ $target_platform == linux-* ]]; then
...
+  export VCPKG_TARGET_TRIPLET="x64-linux"
+fi
+if [[ $target_platform == linux-ppc64le  ]]; then
+  export VCPKG_TARGET_TRIPLET="ppc64le-linux"
+fi
+if [[ $target_platform == linux-aarch64  ]]; then
+  export VCPKG_TARGET_TRIPLET="arm64-linux"
+fi
+
+# For snowflake build only, disabling all the ports:
```
Comment (skupr-anaconda) [inline recipe/build.sh:52]: Does it mean that `abs.yaml` must have `upload_channels: sfe1ed40`? Or will the package be delivered to `defaults`?

File: recipe/meta.yaml [hunk 8]
```diff
@@ -19,32 +17,42 @@ build:
...
+
 requirements:
   build:
-    - make  # [not win]
-    - git   # [not win]
-    - sed   # [not win]
-    - gawk  # [not win]
-    - cmake <=3.21
-    - pkg-config  # [not win]
+    - ninja
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:30]:
```suggestion
    - ninja-base
```
the `base` variant without python

File: recipe/meta.yaml [hunk 9]
```diff
@@ -19,32 +17,42 @@ build:
...
   host:
     - zlib {{ zlib }}
     - bzip2 {{ bzip2 }}
     - zstd {{ zstd }}
-    - lz4-c 1.9.4
-    - curl 7.88.1 # [not win]
-    - libcurl 7.88.1  # [not win]
-    - openssl  {{ openssl }}  # [not win]
+    - lz4-c
+    - libcurl
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:42]:
```suggestion
    - libcurl 8.9.1
```
for an unknown reason, it picks up `"libcurl": "8.1.1"` currently.

File: recipe/meta.yaml (hunk 3 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:25]: `spdlog` relies on `fmt`, see https://metayaml-conda.fly.dev/metayaml/reverse_dependencies?name=fmt

File: recipe/meta.yaml (hunk 6 above)
Comment (Cansisti) [reply recipe/meta.yaml]: done

File: recipe/meta.yaml [hunk 10]
```diff
@@ -19,32 +17,42 @@ build:
...
   run:
+    - fmt  # [win]
     - openssl 3.*
 
 test:
-  #  downstreams:
-  #  - tiledb-py
   commands:
     - test -e $PREFIX/include/tiledb/tiledb.h  # [not win]
     - test -e $PREFIX/include/tiledb/tiledb  # [not win]
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:58]: Could you please also check `Library/lib/pkgconfig/tiledb.pc` on `win` and `lib/pkgconfig/tiledb.pc` on `unix`?

File: recipe/build.sh (hunk 7 above)
Comment (Cansisti) [reply recipe/build.sh:52]: Yes, I will update `abs.yaml` with destination channel.

Comment (Cansisti) [reply recipe/build.sh:52]: This version will only be delivered to sfe1ed40, for defaults we will have to build with all the backends

File: recipe/meta.yaml (hunk 3 above)
Comment (Cansisti) [reply recipe/meta.yaml]: Turned out tiledb needs fmt, and for other platforms it was added implicitly thru spdlog, I've fixed it.

File: recipe/bld.bat (hunk 2 above)
Comment (JeanChristopheMorinPerso) [reply recipe/bld.bat]: Ninja will use all threads by default

File: recipe/build.sh (hunk 7 above)
Comment (Cansisti) [reply recipe/build.sh:52]: done

File: recipe/0002-Don-t-link-to-fmt-library.-We-use-spdlog-s-embedded-.patch (hunk 1 above)
Comment (Cansisti) [reply recipe/0002-Don-t-link-to-fmt-library.-We-use-spdlog-s-embedded-.patch:2]: This patch is unused now, I'll keep the change for now in case Ian was going to use it for something smart

File: recipe/meta.yaml (hunk 10 above)
Comment (Cansisti) [reply recipe/meta.yaml:56]: done

File: recipe/meta.yaml (hunk 4 above)
Comment (Cansisti) [reply recipe/meta.yaml]: I've pinned to the latest as upstream doesn't specify

**Review by skupr-anaconda — APPROVED** [review]

File: recipe/meta.yaml (hunk 5 above)
Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:88]: The selector changed. It was previously `not win` and now it's `win`... The lint should be respected. Please don't add `git` as a dependency on Windows. It'll cause hard to debug issues. The windows instances will have git already installed.

Comment (Cansisti) [reply recipe/meta.yaml:88]: In this case we would need to build libmagic for windows

Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:88]: The instance already has git. Just remove it from the dependencies and you the build system should be able ot use the one installed on the system itself.

Comment (skupr-anaconda) [reply recipe/meta.yaml:88]: Does the linter complain because it expects `git` in the `build` section but `git` is in an unusual `host` section here?

Comment (Cansisti) [reply recipe/meta.yaml:88]: The message states it's just `git` on windows that's an issue. There's another one (warning) that would like to see `git` in `build` and not `host`.
I'll try without `git`, but I believe it was failing in such case.

Comment (skupr-anaconda) [reply recipe/meta.yaml]: An error on `win-64:`
```
CMake Error at build/_deps/vcpkg-src/scripts/buildsystems/vcpkg.cmake:899 (message):

  vcpkg install failed.  See logs for more information:

  C:\b\abs_86yj7i6d8o\croot\tiledb_1731056541602\work\build\vcpkg-bootstrap.log

Call Stack (most recent call first):

  C:/b/abs_86yj7i6d8o/croot/tiledb_1731056541602/_build_env/Library/share/cmake-3.26/Modules/CMakeDetermineSystem.cmake:148 (include)

  CMakeLists.txt:115 (project)

CMake Error: CMake was unable to find a build program corresponding to "Ninja".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.

CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage

CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
```

Comment (Cansisti) [reply recipe/meta.yaml]: Nope, I've tried this too. The message is misleading, the issue lays in `vcpkg install failed` part - that's due to CMake being unable to execute:
```
FetchContent_Declare(
            vcpkg
            GIT_REPOSITORY https://github.com/microsoft/vcpkg.git
            GIT_TAG        ${VCPKG_COMMIT_STRING}
            )
FetchContent_MakeAvailable(vcpkg)
```

File: recipe/meta.yaml [hunk 11]
```diff
@@ -19,42 +19,49 @@ build:
...
-    - git   # [not win]
-    - sed   # [not win]
-    - gawk  # [not win]
-    - cmake <=3.21
-    - pkg-config  # [not win]
+    - m2-patch  # [win]
+    - ninja-base
+    - sed                    # [not win]
+    - gawk                   # [not win]
+    - cmake                  # >=3.21
```
Comment (boldorider4) [inline recipe/meta.yaml:33]: I would either remove the comment or constrain this properly with `cmake >=3.21` if there is reason to do so.

File: recipe/tiledb-patches/system-ports/abseil/vcpkg.json [hunk 12]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (boldorider4) [inline recipe/tiledb-patches/system-ports/abseil/vcpkg.json:1]: can you add an explanation in the PR commit message for all these `vcpkg` related patches?

Comment (Cansisti) [reply recipe/tiledb-patches/system-ports/abseil/vcpkg.json:1]: Honestly, I think Ian was trying something there, but those seem unused at this point.

Comment (boldorider4) [reply recipe/tiledb-patches/system-ports/abseil/vcpkg.json:1]: you are adding a series of files in the `tiledb-patches/system-ports/<pkg-name>/vcpkg.json`. It would be nice to know what those files are about. If you think these files are unused, then you should remove them. Otherwise some documentation in the PR commit message is necessary.

Comment (Cansisti) [reply recipe/tiledb-patches/system-ports/abseil/vcpkg.json:1]: So I spoke with Ian, and here it is:
- Explained in detail here: https://github.com/conda-forge/tiledb-feedstock/tree/main/recipe/tiledb-patches/system-ports
- TLDR: This tells vcpkg to not vendor (download and build) certain dependencies that we have already via conda.
I was wrong claiming those are unused - they are copied by build.* scripts.
They might be needed when we do the full build for defaults (unless we unvendor vcpkg completely). Currently used for windows as we still use vcpkg for libmagic.

I've updated the PR comment with this info.

**Review by JeanChristopheMorinPerso — APPROVED** [review]
Looks good for now. Thanks @ifitchet and @Cansisti!

File: recipe/tiledb-patches/system-ports/abseil/vcpkg.json (hunk 12 above)
Comment (boldorider4) [reply recipe/tiledb-patches/system-ports/abseil/vcpkg.json:1]: awesome! Thank you!

**Review by boldorider4 — APPROVED** [review]
